service-workers/service-worker/resources/fetch-access-control.py

```diff
@@ -1,108 +1,109 @@
-            if split_header[0] == "0" and split_header[1] == "":
+            if split_header[0] == b"0" and split_header[1] == b"":
               end = start
             # Otherwise, it is a range request. Respect the values sent.
-            if split_header[0] != "":
+            if split_header[0] != b"":
               start = int(split_header[0])
-            if split_header[1] != "":
+            if split_header[1] != b"":
               end = int(split_header[1])
           else:
             # The request doesn't have a range. Force a range request by
             # returning the first byte.
             end = start
 
-          headers.append(("Accept-Ranges", "bytes"))
-          headers.append(("Content-Length", str(end -start + 1)))
-          headers.append(("Content-Range", "bytes %d-%d/%d" % (start, end, length)))
+          headers.append((b"Accept-Ranges", b"bytes"))
+          headers.append((b"Content-Length", isomorphic_encode(str(end -start + 1))))
+          headers.append((b"Content-Range", b"bytes %d-%d/%d" % (start, end, length)))
           chunk = body[start:(end + 1)]
           return 206, headers, chunk
         return headers, body
 
-    username = request.auth.username if request.auth.username else "undefined"
-    password = request.auth.password if request.auth.username else "undefined"
-    cookie = request.cookies['cookie'].value if 'cookie' in request.cookies else "undefined"
+    username = request.auth.username if request.auth.username else b"undefined"
+    password = request.auth.password if request.auth.username else b"undefined"
+    cookie = request.cookies[b'cookie'].value if b'cookie' in request.cookies else b"undefined"
 
     files = []
-    for key, values in request.POST.iteritems():
+    for key, values in request.POST.items():
         assert len(values) == 1
         value = values[0]
-        if not hasattr(value, "file"):
+        if not hasattr(value, u"file"):
             continue
         data = value.file.read()
-        files.append({"key": key,
-                      "name": value.file.name,
-                      "type": value.type,
-                      "error": 0, #TODO,
-                      "size": len(data),
-                      "content": data})
+        files.append({u"key": isomorphic_decode(key),
+                      u"name": value.file.name,
+                      u"type": value.type,
+                      u"error": 0, #TODO,
+                      u"size": len(data),
+                      u"content": data})
 
-    get_data = {key:request.GET[key] for key,value in request.GET.iteritems()}
-    post_data = {key:request.POST[key] for key,value in request.POST.iteritems()
-                 if not hasattr(request.POST[key], "file")}
-    headers_data = {key:request.headers[key] for key,value in request.headers.iteritems()}
+    get_data = {isomorphic_decode(key):isomorphic_decode(request.GET[key]) for key, value in request.GET.items()}
```

Comment:
All of the `isomorphic_decode` makes this a bit hard to read now. Should we have a helper like `isomorphic_decode(request.GET)` which converts all of the keys and values to Unicode strings? This need not block this PR, however.

@Hexcles for opinions.

Reply:
I see that @Hexcles is OOO. As I said this shouldn't block this PR, so I'll just leave this comment here.